From 9934901a73ea51b827f3ad45252874912a4790d9 Mon Sep 17 00:00:00 2001
From: Chris <chris@echoz.io>
Date: Sun, 4 Oct 2026 16:23:32 +0000
Subject: [PATCH 2/2] reload: use SIGUSR1 instead of SIGHUP

SIGHUP is sent by the terminal when it closes. Handling it for :reload
leaves aerc running as an orphan after its terminal goes away
(https://todo.sr.ht/~rjarry/aerc/355). Use SIGUSR1 for reload and let
SIGHUP terminate the process as before.
---
 CHANGELOG.md       | 2 +-
 commands/reload.go | 4 ++--
 doc/aerc.1.scd     | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/CHANGELOG.md b/CHANGELOG.md
index 2893a129..42e87789 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,7 +7,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ### Added
 
-- `:reload` is now invoked when `SIGHUP` is received by the `aerc` process.
+- `:reload` is now invoked when `SIGUSR1` is received by the `aerc` process.
 - Add support for Labels when using imap.
 - A custom directory can now be specified for temporary files.
 - New `:toggle-sidebar` command to hide/show the sidebar.
diff --git a/commands/reload.go b/commands/reload.go
index 43319c0b..80936542 100644
--- a/commands/reload.go
+++ b/commands/reload.go
@@ -20,11 +20,11 @@ type Reload struct {
 func init() {
 	Register(Reload{})
 	sig := make(chan os.Signal, 1)
-	signal.Notify(sig, syscall.SIGHUP)
+	signal.Notify(sig, syscall.SIGUSR1)
 	go func() {
 		defer log.PanicHandler()
 		for range sig {
-			log.Infof("received SIGHUP, reloading configuration")
+			log.Infof("received SIGUSR1, reloading configuration")
 			err := Reload{}.Execute(nil)
 			if err != nil {
 				log.Errorf("reload failed: %s", err)
diff --git a/doc/aerc.1.scd b/doc/aerc.1.scd
index 2508ee00..e0167b29 100644
--- a/doc/aerc.1.scd
+++ b/doc/aerc.1.scd
@@ -296,7 +296,7 @@ _<aerc-cmd ...>_
 	*-s* _<styleset-name>_
 		Load the specified styleset.
 
-	Sending the *SIGHUP* signal to the *aerc* process will cause *:reload*
+	Sending the *SIGUSR1* signal to the *aerc* process will cause *:reload*
 	to be performed (with no flags).
 
 *:repeat*
-- 
2.55.0

